libmachine/provision/redhat.go

```diff
@@ -0,0 +1,279 @@
+
+	if err := provisioner.SetHostname(provisioner.Driver.GetMachineName()); err != nil {
+		return err
+	}
+
+	// setup extras repo
+	if err := provisioner.configureRepos(); err != nil {
+		return err
+	}
+
+	for _, pkg := range provisioner.packages {
+		if err := provisioner.Package(pkg, pkgaction.Install); err != nil {
+			return err
+		}
+	}
+
+	// install docker
+	if err := provisioner.installDocker(); err != nil {
+		return err
+	}
+
+	if err := utils.WaitFor(provisioner.dockerDaemonResponding); err != nil {
+		return err
+	}
+
+	if err := makeDockerOptionsDir(provisioner); err != nil {
+		return err
+	}
+
+	provisioner.AuthOptions = setRemoteAuthOptions(provisioner)
+
+	if err := ConfigureAuth(provisioner); err != nil {
+		return err
+	}
+
+	if err := configureSwarm(provisioner, swarmOptions); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (provisioner *RedHatProvisioner) Hostname() (string, error) {
+	output, err := provisioner.SSHCommand("hostname")
+	if err != nil {
+		return "", err
+	}
+
+	var so bytes.Buffer
+	if _, err := so.ReadFrom(output.Stdout); err != nil {
+		return "", err
+	}
+
+	return so.String(), nil
+}
+
+func (provisioner *RedHatProvisioner) SetHostname(hostname string) error {
+	if out, err := provisioner.SSHCommand(fmt.Sprintf(
+		"sudo hostname %s && echo %q | sudo tee /etc/hostname",
+		hostname,
+		hostname,
+	)); err != nil {
+		log.Info(out)
+		log.Errorf("error setting hostname: %s", err)
+		return err
+	}
+
+	if _, err := provisioner.SSHCommand(fmt.Sprintf(
+		"if grep -xq 127.0.1.1.* /etc/hosts; then sudo sed -i 's/^127.0.1.1.*/127.0.1.1 %s/g' /etc/hosts; else echo '127.0.1.1 %s' | sudo tee -a /etc/hosts; fi",
+		hostname,
+		hostname,
+	)); err != nil {
+		log.Errorf("error setting /etc/hosts: %s", err)
+		return err
+	}
+
+	return nil
+}
+
+func (provisioner *RedHatProvisioner) GetDockerOptionsDir() string {
+	return "/etc/docker"
+}
+
+func (provisioner *RedHatProvisioner) SSHCommand(args string) (ssh.Output, error) {
+	return drivers.RunSSHCommandFromDriver(provisioner.Driver, args)
+}
+
+func (provisioner *RedHatProvisioner) CompatibleWithHost() bool {
+	return provisioner.OsReleaseInfo.Id == "rhel"
+}
+
+func (provisioner *RedHatProvisioner) GetAuthOptions() auth.AuthOptions {
+	return provisioner.AuthOptions
+}
+
+func (provisioner *RedHatProvisioner) SetOsReleaseInfo(info *OsRelease) {
+	provisioner.OsReleaseInfo = info
+}
+
+func (provisioner *RedHatProvisioner) GenerateDockerOptions(dockerPort int) (*DockerOptions, error) {
+	var (
+		engineCfg  bytes.Buffer
+		configPath = "/etc/sysconfig/docker"
+	)
+
+	// remove existing
+	if _, err := provisioner.SSHCommand(fmt.Sprintf("sudo rm %s", configPath)); err != nil {
+		return nil, err
+	}
+
+	driverNameLabel := fmt.Sprintf("provider=%s", provisioner.Driver.DriverName())
+	provisioner.EngineOptions.Labels = append(provisioner.EngineOptions.Labels, driverNameLabel)
+
+	// systemd / redhat will not load options if they are on newlines
```

Comment:
Yeah, we should be able to generate just a simplified stub like:
```
[Service]
ExecStart=
ExecStart=/usr/bin/docker -d <whatever flags we need to add here>
```

It might be safer if we figure out some way to ask where the existing `docker.service` file is so we can scrape the `ExecStart` line it has and not break things like socket activation, but I don't know how much that really matters in this case either.  Worth playing around for sure.